### PR TITLE
fix: detecting multiple episodes in symlink library

### PR DESCRIPTION
**Pull request by FlashLightning67**
Ensures that all episodes are found from symlink titles when ingesting symlinks into the db. Fixes an issue with symlink ingestion where files containing multiple episodes would only have one of the episodes added to the db, which causes the missing episodes to be submitted for scraping again after every db reset.

## Summary by CodeRabbit

- **New Features**
	- Enhanced episode processing to handle multiple episodes from filenames.
	- Improved error handling for missing episode information with automatic file deletion.

- **Bug Fixes**
	- Refined logic for extracting episode numbers, leading to more reliable indexing.

- **Chores**
	- Minor adjustments to logging statements for consistency.

### CODE REVIEW

**Comment by dreulavelle** [comment]
Have you tested this? We also need to verify that the downloaders are downloading episodes appropriately too. e.g. settings the same file for both episode items.

File: src/program/services/libraries/symlink.py [hunk 1]
```diff
@@ -159,24 +159,29 @@ def process_shows(directory: Path, item_type: str, is_anime: bool = False) -> Ge
             for episode in os.listdir(directory / show / season):
                 if os.path.splitext(episode)[1][1:] not in ALLOWED_VIDEO_EXTENSIONS:
                     continue
-                if not (episode_number := re.search(r"s\d+e(\d+)", episode, re.IGNORECASE)):
+                episode_numbers = re.findall(r"s\d+e(\d+)(?:-e(\d+))?", episode, re.IGNORECASE)
```
Comment (Gaisberg) [inline src/program/services/libraries/symlink.py:162]: Now that I think of it, please use RTN/parsett for this.

**Review by Gaisberg — CHANGES_REQUESTED** [review]

File: src/program/services/libraries/symlink.py (hunk 1 above)
Comment (dreulavelle) [reply src/program/services/libraries/symlink.py:162]: use parsett, it'll be faster to process.

heres an example

```py
from PTT import parse_title

episodes: list[int] = parse_title(episode).episodes
```

File: src/program/services/libraries/symlink.py [hunk 2]
```diff
@@ -159,24 +160,26 @@ def process_shows(directory: Path, item_type: str, is_anime: bool = False) -> Ge
...
-                    episode_item.set("symlinked", True)
-                    episode_item.set("update_folder", str(Path(directory) / show / season / episode))
-                else:
-                    episode_item.set("symlinked", True)
-                    episode_item.set("update_folder", "updated")
-                if is_anime:
-                    episode_item.is_anime = True
-                episodes[int(episode_number.group(1))] = episode_item
+                for episode_number in episode_numbers:
+                    episode_item = Episode({"number": episode_number})
```
[the remark this answers is not in the document]
Comment (dreulavelle) [reply src/program/services/libraries/symlink.py:171]: its always an int, or its empty. this isnt necessary.

File: src/program/services/libraries/symlink.py [hunk 3]
```diff
@@ -159,24 +160,26 @@ def process_shows(directory: Path, item_type: str, is_anime: bool = False) -> Ge
             for episode in os.listdir(directory / show / season):
                 if os.path.splitext(episode)[1][1:] not in ALLOWED_VIDEO_EXTENSIONS:
                     continue
-                if not (episode_number := re.search(r"s\d+e(\d+)", episode, re.IGNORECASE)):
+                episode_numbers: list[int] = parse_title(episode).get("episodes", [])
+                if not episode_numbers:
                     logger.log("NOT_FOUND", f"Can't extract episode number at path {directory / show / season / episode}")
                     # Delete the episode since it can't be indexed
                     os.remove(directory / show / season / episode)
                     continue
```
[the remark this answers is not in the document]
Comment (dreulavelle) [reply src/program/services/libraries/symlink.py:168]: not needed.

**Review by dreulavelle — APPROVED** [review]
This looks good now!

File: src/program/services/libraries/symlink.py [hunk 4]
```diff
@@ -159,24 +159,29 @@ def process_shows(directory: Path, item_type: str, is_anime: bool = False) -> Ge
...
-                    episode_item.set("symlinked", True)
-                    episode_item.set("update_folder", "updated")
-                if is_anime:
-                    episode_item.is_anime = True
-                episodes[int(episode_number.group(1))] = episode_item
+                for ep_range in episode_numbers:
+                    start_ep = int(ep_range[0])
+                    end_ep = int(ep_range[1]) if ep_range[1] else start_ep
+                    for episode_number in range(start_ep, end_ep + 1):
+                        episode_item = Episode({"number": episode_number})
```
[the remark this answers is not in the document]
Comment (dreulavelle) [reply src/program/services/libraries/symlink.py]: We dont care about index as we symlink from low > high anyways. So this isn't needed.

**Review by dreulavelle — APPROVED** [review]

**Pull request merged**